### ISSUE TITLE
Building harfbuzz fails with --enable-introspection

### ISSUE THREAD

**Issue by MarekSchiffer** [harfbuzz/harfbuzz#2081]
Hello, I hope this is the right place to ask. I'm trying to build harfbuzz with introspection i.e. 

> ./configure --prefix=/opt/ --enable-static --with-coretext  --with-graphite2 --with-gobject 
> **--enable-introspection**

If I build it with:
> ./configure --prefix=/opt/ --enable-static --with-coretext  --with-graphite2 --with-gobject 
> **--enable-introspection=no** 

it works fine.
Compiling fails with the message:
> dyld: Library not loaded: @rpath/libgraphite2.3.dylib
>   Referenced from: /Users/schiffer/Downloads/harfbuzz-2.6.4/src/tmp-introspecti1g7bvmw/.libs/HarfBuzz-0.0
>   Reason: image not found
> Command '['/Users/schiffer/Downloads/harfbuzz-2.6.4/src/tmp-introspecti1g7bvmw/HarfBuzz-0.0', '--introspect-dump=/Users/schiffer/Downloads/harfbuzz-2.6.4/src/tmp-introspecti1g7bvmw/functions.txt,/Users/schiffer/Downloads/harfbuzz-2.6.4/src/tmp-introspecti1g7bvmw/dump.xml']' died with <Signals.SIGABRT: 6>.
> make[4]: *** [HarfBuzz-0.0.gir] Error 1
> make[3]: *** [all-recursive] Error 1
> make[2]: *** [all] Error 2
> make[1]: *** [all-recursive] Error 1
> make: *** [all] Error 2'

I have libgraphite2.3.dylib installed in /opt/lib and linked to /urs/local/lib as well.

otool -L /opt/lib/libgraphite2.3.dylib  gives me

>  /opt/lib/libgraphite2.3.dylib:
> 	@rpath/libgraphite2.3.dylib (compatibility version 3.0.0, current version 3.2.1)
> 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.0.0)

I suspect that the rpath gets messed up but at this point I'm completely clueless
Any help would be highly appreciated!

**Comment by behdad** [comment]
It's def not your fault.  Is interaction of Mac linker and gobject-introspection build system and libtool and all..   Umm. Am not sure how to debug / what to suggest.

**Comment by MarekSchiffer** [comment]
Hello, 
thank you for your respond, although it was quite devastating at the time and I relied on Homebrew.

I recently tried it again, and the same error occurred, now macOS Catalina 10.15.6.
The solution turned out to be quite simple, 

first I had to add

`if (APPLE) `
`set(CMAKEMACOSXRPATH ON)`
`endif (APPLE)`

to CMakeLists.txt of Graphite2,
secondly the LDFLAGS had to be adjusted as follows:

`export LDFLAGS=”−Wl,−rpath,/PATH_TO/graphite2−1.3.14/lib”` .

After that everything worked fine.
I hope that helps somebody.
